src/util/windows.jl

```diff
@@ -0,0 +1,53 @@
+"""
+An iterator which moves across a container, as a sliding window.
+"""
+immutable EachWindowIterator{T <: AbstractArray}
+    "A reference to the collection to iterate over."
+    data::T
+    "The width of the sliding window."
+    width::Int
+    "How many positions the sliding window moves along."
+    step::Int
+    "The number of windows that will result from iterating across the container."
+    nwin::Int
+    "The number of elements in the container that are missed. Typically because
+    of the width and step size of the window: some elements near the end of the
+    container are missed if window sizes and step sizes are too large."
+    nmissed::Int
+end
```

Comment:
The Julia documentation says you can attach docstrings to fields:
"
```
"..."
type T
    "x"
    x
    "y"
    y
end
```
Adds docstring "..." to type T, "x" to field T.x and "y" to field T.y. Also applicable to immutable types."